### PR TITLE
Add FormatConfig.WithComplexPlugin for safe plugin registration

### PR DESCRIPTION
**Pull request by apstndb**
## Summary

- Add `(*FormatConfig).WithComplexPlugin` to clone a config and append one `FormatComplexFunc` without mutating shared preset singletons.
- Document the helper in package, `FormatConfig`, and `doc.go` godoc.
- Add tests for nil receiver/plugin rejection, preset isolation, chaining, and clone independence.

Closes #181.

## Test plan

- [x] `make check`
- [x] `go test ./... -run TestFormatConfigWithComplexPlugin`

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: format_config.go [hunk 1]
```diff
@@ -45,6 +45,19 @@ func hasPresetScalarPlugin(fc *FormatConfig) bool {
 	return slices.ContainsFunc(fc.FormatComplexPlugins, isPresetScalarPlugin)
 }
 
+// WithComplexPlugin returns a clone of fc with plugin appended to FormatComplexPlugins.
+// The original config, including shared preset singletons, is not mutated. Chain
+// further calls on the returned config for additional plugins. Nil fc or nil plugin
+// returns nil.
+func (fc *FormatConfig) WithComplexPlugin(plugin FormatComplexFunc) *FormatConfig {
+	if fc == nil || plugin == nil {
```
Comment (apstndb) [inline format_config.go]: Returning nil when only `plugin` is nil (but `fc` is valid) silently discards an otherwise-good config, and because the method is designed for chaining, a single nil anywhere in `a.WithComplexPlugin(p1).WithComplexPlugin(nil).WithComplexPlugin(p2)` collapses the whole chain to nil and the final `.FormatColumn` panics far from the cause. Nil `fc` -> nil is an established convention here (`Clone`, `FormatConfigWithoutScalarPlugins`), but a nil *plugin* is a distinct programmer error. Worth considering whether `panic` (fail-fast at the call site) is preferable for the nil-plugin case, or at least calling out the chain hazard explicitly in the godoc. Returning `fc` unchanged is tempting but unsafe, since callers would then mutate what they think is a clone.

**Review by apstndb — COMMENTED** [review]
Useful ergonomic helper, and the clone-then-append approach correctly avoids mutating shared preset singletons. Tests cover nil receiver/plugin, preset isolation, chaining, and element-level clone independence. One design point on the nil-plugin behavior inline.

File: common.go [hunk 2]
```diff
@@ -229,7 +229,8 @@ type Formatter interface {
 //     [ErrFormatNullableRequired] without Decode. NULL scalars use [*FormatConfig.GetNullString]
 //     before the slow path runs (FormatNullable is not called for NULL).
 //
-// Use [FormatConfig.Clone] to customize a preset without mutating shared instances.
+// Use [FormatConfig.Clone] or [FormatConfig.WithComplexPlugin] to customize a preset
+// without mutating shared instances.
```
[the remark this answers is not in the document]
Comment (apstndb) [reply common.go]: Updated common.go links to \`[*FormatConfig.Clone]\` and \`[*FormatConfig.WithComplexPlugin]\`.

File: doc.go [hunk 3]
```diff
@@ -18,8 +18,9 @@
 // [FormatConfig.FormatComplexPlugins] to use Decode + [FormatConfig.FormatNullable]
 // (set FormatNullable on the clone; nil returns [ErrFormatNullableRequired]).
 // Scalar plugins fall through to that path when [FormatConfig.FormatNullable] is set.
-// Constructors return a new [FormatConfig]; call [FormatConfig.Clone] before mutating
-// a config you may reuse ([FormatConfig.Clone] copies [FormatConfig.FormatComplexPlugins]).
+// Constructors return a new [FormatConfig]; call [FormatConfig.Clone] or
+// [FormatConfig.WithComplexPlugin] before mutating a config you may reuse
+// ([FormatConfig.Clone] copies [FormatConfig.FormatComplexPlugins]).
```
[the remark this answers is not in the document]
Comment (apstndb) [reply doc.go]: Updated package doc links to use pointer-receiver \`*FormatConfig\` method forms.

File: format_config.go (hunk 1 above)
Comment (apstndb) [reply format_config.go]: Nil plugin now panics at the call site (nil receiver still returns nil). Godoc calls out the chain hazard; test updated.

**Review by apstndb — COMMENTED** [review]
Re-review of the new commit (c2a44f2 -> 1d47ed4). My design point is resolved: `WithComplexPlugin` now panics on a nil plugin (`panic("spanvalue: WithComplexPlugin: nil plugin")`) while keeping nil `fc` -> nil, so a mistaken nil fails at the call site instead of silently collapsing a chain to nil. Godoc updated to state this, and the test now asserts the panic. Pointer-receiver godoc links also fixed. Correct and matches the suggestion. LGTM.
